packages/amplify-category-hosting/lib/S3AndCloudFront/helpers/file-uploader.js

```diff
@@ -6,20 +6,36 @@ const sequential = require('promise-sequential');
 const fileScanner = require('./file-scanner');
 const constants = require('../../constants');
 
+const category = 'hosting';
 const serviceName = 'S3AndCloudFront';
 const providerName = 'awscloudformation';
 
 async function run(context, distributionDirPath) {
   const { WebsiteConfiguration } = context.exeInfo.template.Resources.S3Bucket.Properties;
+  const { output } = context.exeInfo.amplifyMeta[category][serviceName];
   const fileList =
-    fileScanner.scan(context, distributionDirPath, WebsiteConfiguration.IndexDocument);
+    fileScanner.scan(context, distributionDirPath, WebsiteConfiguration);
 
   const uploadFileTasks = [];
   const s3Client = await getS3Client(context, 'update');
   const hostingBucketName = getHostingBucketName(context);
+
+  let cloudFrontS3CanonicalUserId;
+  if (context.exeInfo.template.Resources.OriginAccessIdentity &&
```

Comment:
Is context.exeInfo.template.Resources.OriginAccessIdentity always defined?

Reply:
No, only when CloudFront is enabled. It is deleted when CloudFront is disabled.